subprojects/docs/src/docs/userguide/build-cache/build_cache_debugging.adoc

```diff
@@ -196,91 +196,4 @@ Build cache key for task ':compileJava' is 8ebf682168823f662b9be34d27afdf77
-for the task class itself and for each of its actions. If there is a change, this means that the build script, `buildSrc` or the Gradle version has changed.
-
-[NOTE]
-====
-A change in the output of `buildSrc` also marks all the logic added by your build as changed.
-Especially, custom actions added to cacheable tasks will be marked as changed.
-This can be problematic, see <<common_caching_problems.adoc#custom_actions, section about `doFirst` and `doLast`>>.
-====
-
-If the implementation is the same, then you need to start comparing inputs between the two builds.
-There should be at least one different input hash. If it is a simple value property, then the configuration of the task changed.
-This can happen for example by
-
- * changing the build script,
- * conditionally configuring the task differently for CI or the developer builds,
- * depending on a system property or an environment variable for the task configuration,
- * or having an absolute path which is part of the input.
-
-If the changed property is a file property, then the reasons can be the same as for the change of a value property.
-Most probably though a file on the filesystem changed in a way that Gradle detects a difference for this input.
-The most common case will be that the source code was changed by a check in.
-It is also possible that a file generated by a task changed, e.g. since it includes a timestamp.
-As described in <<common_caching_problems.adoc#java_version_tracking,Java version tracking>>, the Java version can also influence the output of the Java compiler.
-If you did not expect the file to be an input to the task, then it is possible that you should alter the configuration of the task to not include it.
-For example, having your integration test configuration including all the unit test classes as a dependency has the effect that all integration tests
-are re-executed when a unit test changes.
-Another option is that the task tracks absolute paths instead of relative paths and the location of the project directory changed on disk.
-
-== Example
-
-We will walk you through the process of diagnosing a cache miss.
-Let's say we have build `A` and build `B` and we expected all the test tasks for a sub-project `sub1` to be cached in build `B` since only a unit test for another sub-project `sub2` changed.
-Instead, all the tests for the sub-project have been executed.
-Since we have the cascading effect when we have cache misses, we need to find the task which caused the caching chain to fail.
-This can easily be done by filtering for all cacheable tasks which have been executed and then select the first one.
-In our case, it turns out that the tests for the sub-project `internal-testing` were executed even though there was no code change to this project.
-We start the input property comparison in Gradle Enterprise and see that the property `classpath` changed. This means that some file on the runtime classpath actually did change.
```

Comment:
> We start the input property comparison in Gradle Enterprise and see that the property `classpath` changed. This means that some file on the runtime classpath actually did change.

This should be removed. I'm not sure if it causes too much of a leap to the next line.